doc/source/conf.py

```diff
@@ -302,6 +300,9 @@
 plot_pre_code = """
 import warnings
 for key in (
+        'lsim2 is deprecated',  # Deprecation of scipy.signal.lsim2
+        'impulse2 is deprecated',  # Deprecation of scipy.signal.impulse2
+        'step2 is deprecated',  # Deprecation of scipy.signal.step2
```

Comment:
OK, I see you need to (perhaps) add something here, but why all the other changes in this file?

I say "perhaps" because it seems to me that we don't need to add this to _every single code invocation_, but rather just those where one of the deprecated methods gets called, or better: change those call-sites.

Reply:
>  or better: change those call-sites.
I don't think this is possible in this circumstance. All the functions add here are the once being deprecated; it is impossible to avoid their own examples from generated a deprecation warning. I imagine the other changes relate to the recent changes in lint infrastructure

Reply:
> it is impossible to avoid their own examples from generated a deprecation warning.

We could remove their examples? Serious question

Reply:
Maybe? I am not sure it is worth it; they will get removed eventually anyway and there is no precedent or policy for doing so. I understand your hesitance about heavy handed filtering of warning but I think it is fine as it is only for the doc build, any unintended use of these deprecated functions would come out in the test suite.

Reply:
> OK, I see you need to (perhaps) add something here, but why all the other changes in this file?

The other changes were done due to the new linter configuration.